.pre-commit-config.yaml

```diff
@@ -25,12 +24,13 @@ repos:
       - id: check-copyright
         args: ['--config', '.ci/check_copyright_config.yaml']
 
-  - repo: https://github.com/pylint-dev/pylint
-    rev: v3.0.3
+  - repo: local
     hooks:
       - id: pylint                                       # Runs pylint on Python code
-        additional_dependencies: ['PyYAML']
-        args: ['--rcfile=.ci/.pylintrc']
+        name: pylint
+        entry: sh -s "export PYTHONPATH=$(pwd)/ksz8863/."
```

Comment:
This is not nice. Is it connected with https://github.com/espressif/esp-eth-drivers/pull/61/files#r2303729871? I wanted to keep `pylint` but it doesn't make sense if we must make such workarounds. I checked and IDF doesn't use it, please remove it too. (And all associated pylint configuration).